Cards/CardBase.gd

```diff
@@ -20,6 +20,8 @@ class_name CardBase
 
 @export var card_effects_data: Array[EffectData] = []
 
+@export var energy: EnergyData = EnergyData.new()
```

Comment:
I would maybe name this something else than `energy`
because `energy` alone feels like this would be a cost (which is in fact `energy_cost`)
Maybe `energy_info` would be more fitting (so we know it's information and we need to query the actual things in it)